src/GraphQL.Tests/Serialization/DeserializationTestBase.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Numerics;
+using Shouldly;
+
+namespace GraphQL.Tests.Serialization
+{
+    public class DeserializationTestBase
+    {
+        protected readonly TestData ExampleData = new TestData
+        {
+            array = new object[]
+                {
+                    null,
+                    "test",
+                    123,
+                    1.2
+                },
+            obj = new TestChildData
+            {
+                itemNull = null,
+                itemString = "test",
+                itemNum = 123,
+                itemFloat = 12.4,
+            },
+            itemNull = null,
+            itemString = "test",
+            itemNum = 123,
+            itemFloat = 12.4,
+            itemBigInt = BigInteger.Parse("1234567890123456789012345678901234567890"),
+        };
+        protected readonly string ExampleJson = "{\"array\":[null,\"test\",123,1.2],\"obj\":{\"itemNull\":null,\"itemString\":\"test\",\"itemNum\":123,\"itemFloat\":12.4},\"itemNull\":null,\"itemString\":\"test\",\"itemNum\":123,\"itemFloat\":12.4,\"itemBigInt\":1234567890123456789012345678901234567890}";
+
+        public class TestData
+        {
+            public object[] array { get; set; }
+            public TestChildData obj { get; set; }
+            public string itemNull { get; set; }
+            public string itemString { get; set; }
+            public int itemNum { get; set; }
+            public double itemFloat { get; set; }
+            public BigInteger itemBigInt { get; set; }
+        }
+
+        public class TestChildData
+        {
+            public string itemNull { get; set; }
+            public string itemString { get; set; }
+            public int itemNum { get; set; }
+            public double itemFloat { get; set; }
+        }
+
+        protected void Verify(IReadOnlyDictionary<string, object> actual)
+        {
+            var array = actual["array"].ShouldBeOfType<List<object>>();
+            array[0].ShouldBeNull();
+            array[1].ShouldBeOfType<string>().ShouldBe("test");
+            array[2].ShouldBeOfType<int>().ShouldBe(123);
+            array[3].ShouldBeOfType<double>().ShouldBe(1.2);
+            var obj = actual["obj"].ShouldBeOfType<Dictionary<string, object>>();
+            obj["itemNull"].ShouldBeNull();
+            obj["itemString"].ShouldBeOfType<string>().ShouldBe("test");
+            obj["itemNum"].ShouldBeOfType<int>().ShouldBe(123);
+            obj["itemFloat"].ShouldBeOfType<double>().ShouldBe(12.4);
+            actual["itemNull"].ShouldBeNull();
+            actual["itemString"].ShouldBeOfType<string>().ShouldBe("test");
+            actual["itemNum"].ShouldBeOfType<int>().ShouldBe(123);
+            actual["itemFloat"].ShouldBeOfType<double>().ShouldBe(12.4);
+            actual["itemBigInt"].ShouldBeOfType<BigInteger>().ShouldBe(ExampleData.itemBigInt);
```

Comment:
You compare actual with ExampleData here but not in all other lines. Why?

Reply:
Can't remember.  This is an old test, although I think I did write it a long time ago.  Probably because it's not really possible to write a BigInteger literal.  If you want me to copy `BigInteger.Parse("1234567890123456789012345678901234567890")` in here, I can... (lmk)

Reply:
I thought to offer the opposite - use properties from ExampleData everywhere.

Reply:
🤷‍♂️ ok done.